mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java

```diff
@@ -126,7 +126,7 @@ public void save(MailboxMessage mailboxMessage) throws MailboxException {
         mailboxMapper.findMailboxById(mailboxId);
         ComposedMessageIdWithMetaData composedMessageIdWithMetaData = createMetadataFor(mailboxMessage);
         messageDAO.save(mailboxMessage)
-            .thenMany(Flux.merge(
+            .thenMany(Flux.concat(
                 imapUidDAO.insert(composedMessageIdWithMetaData),
                 messageIdDAO.insert(composedMessageIdWithMetaData)))
```

Comment:
as I understand the `messageIdDAO` is the source and `imapUidDAO` is the projection. Why saving saves the projection first, then the source? I though it should be in a reverse way

Reply:
It's the other way around: that's not what I intend to fix here anyway.

Cf https://github.com/linagora/james-project/pull/3148/commits/325e79b1e6f72a96798c0f0de153b5ee70fa2c37